src/state/service.rs

```diff
@@ -80,37 +81,38 @@ impl<'de> serde::Deserialize<'de> for EndpointSet {
     where
         D: Deserializer<'de>,
     {
-        <HashMap<Strng, Endpoint>>::deserialize(deserializer).map(|inner| EndpointSet { inner })
+        <HashMap<Strng, Arc<Endpoint>>>::deserialize(deserializer)
+            .map(|inner| EndpointSet { inner })
     }
 }
 
 impl EndpointSet {
     pub fn from_list<const N: usize>(eps: [Endpoint; N]) -> EndpointSet {
         let mut endpoints = HashMap::with_capacity(eps.len());
         for ep in eps.into_iter() {
-            endpoints.insert(ep.workload_uid.clone(), ep);
+            endpoints.insert(ep.workload_uid.clone(), Arc::new(ep));
         }
         EndpointSet { inner: endpoints }
     }
 
     pub fn insert(&mut self, k: Strng, v: Endpoint) {
-        self.inner.insert(k, v);
+        self.inner.insert(k, Arc::new(v));
     }
 
     pub fn contains(&self, key: &Strng) -> bool {
         self.inner.contains_key(key)
     }
 
     pub fn get(&self, key: &Strng) -> Option<&Endpoint> {
-        self.inner.get(key)
+        self.inner.get(key).map(|e| &**e)
```

Comment:
```suggestion
        self.inner.get(key).map(Arc::as_ref)
```

Reply:
Nice, you are better than chatgpt :slightly_smiling_face:

Reply:
tbh, it looked sooooo crazy to me that I spent a likely shameful amount of time trying to understand how rustc doesn't alert the FBI when this tries to compile. After that it was "oh, isn't this the same as?" 😂